docs/development/authentication.md

```diff
@@ -0,0 +1,137 @@
+# Understanding authentication at runtime
+
+The following document provides an introduction around the different authentication methods that can take place during an image build when using the Build operator.
+
+- [Overview](#overview)
+- [Authentication for Git](#authentication-for-git)
+  - [Basic authentication](#basic-authentication)
+  - [SSH authentication](#ssh-authentication)
+  - [Usage of git secret](#usage-of-git-secret)
+- [Authentication to container registries](#authentication-to-container-registries)
+  - [Docker Hub](#docker-hub)
+  - [Usage of registry secret](#usage-of-registry-secret)
+- [References](#references)
+
+## Overview
+
+There are two places where users might need to define authentication when building images. Authentication to a container registry is the most common one, but also users might have the need
+to define authentications for pulling source-code from Git.
+
+## Authentication for Git
+
+There are two ways for authenticating into Git(_applies to both Gitlab or Github_). Also, see more information in the official Tekton [docs](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#configuring-ssh-auth-authentication-for-git).
+
+### SSH authentication
+
+For the ssh authentication you can use the tekton annotations to specify if you are cloning from github or gitlab. Is recommended to always use both when defining your secret.
```

Comment:
Hm, I am thinking about if that recommendation is good - I feel a little uncomfortable from a security perspective as one does not want one's private key to be used for undesired purposes. What about this as an alternative:

*For the SSH authentication you must use the tekton annotations to specify the hostname(s) of the git repository providers that you use. This is `github.com` for GitHub, or `gitlab.com` for GitLab, but can also be the host name of the GitHub Enterprise instance of your company.*

Reply:
I like that suggestion.

Reply:
Thanks. Much better. I will not write the github enterprise one because I never tested it, but please enhance the document with it later if you did.

> I feel a little uncomfortable from a security perspective as one does not want one's private key to be used for undesired purposes.

This is true, but also git´s allow you to generate multiple ssh keys, so you can use one and delete it later. This is supported in tekton, therefore we need to clearly communicate to users.

Not sure if I fully understand you, I thnínk

Reply:
I tried it with the IBM GHE and can confirm that it works. Recommendation to add both is gone, that's the important part.